src/abstractdataframe/abstractdataframe.jl

```diff
@@ -573,27 +574,29 @@ julia> completecases(df, [:x, :y])
 ```
 
 """
-function completecases(df::AbstractDataFrame)
+function completecases(df::AbstractDataFrame, col::Colon=:)
+    if ncol(df) == 0
+        throw(ArgumentError("Unable to compute complete cases of a data frame with no columns"))
+    end
     res = trues(size(df, 1))
     for i in 1:size(df, 2)
         _nonmissing!(res, df[i])
     end
     res
 end
 
-function completecases(df::AbstractDataFrame, col::Union{Integer, Symbol})
+function completecases(df::AbstractDataFrame, col::ColumnIndex)
     res = trues(size(df, 1))
     _nonmissing!(res, df[col])
     res
 end
 
-completecases(df::AbstractDataFrame, cols::Union{AbstractVector, Regex}) =
-    completecases(df[cols])
+completecases(df::AbstractDataFrame, cols::Union{AbstractVector, Regex, Not}) =
+    completecases(select(df, cols, copycols=false))
 
 """
     dropmissing(df::AbstractDataFrame; disallowmissing::Bool=true)
-    dropmissing(df::AbstractDataFrame, cols::Union{AbstractVector, Regex}; disallowmissing::Bool=true)
-    dropmissing(df::AbstractDataFrame, cols::Union{Integer, Symbol}; disallowmissing::Bool=true)
+    dropmissing(df::AbstractDataFrame, cols; disallowmissing::Bool=true)
```

Comment:
`cols` should be described a bit in the text. I think we have a standard expression for values that can be used with indexing.

Also, the types are listed for `completecases`, so we should be consistent.

Same for `dropmissing!`.

Reply:
fixed